bc_obps/reporting/service/report_supplementary_version_service.py

```diff
@@ -0,0 +1,537 @@
+)
+
+
+class ReportSupplementaryVersionService:
+    @staticmethod
+    @transaction.atomic
+    def create_report_supplementary_version(report_version_id: int) -> ReportVersion:
+        """
+        Creates a new report version for a given report version ID cloning related data.
+
+        Args:
+            report_version_id: The report version ID
+
+        Returns:
+            New ReportVersion instance
+        """
+        # Get the existing report version
+        report_version = ReportVersion.objects.get(id=report_version_id)
+        # Create a new report version as a Draft
+        new_report_version = ReportVersion.objects.create(
+            report=report_version.report,
+            report_type=report_version.report_type,
+            status=ReportVersion.ReportVersionStatus.Draft,
+            is_latest_submitted=False,
+        )
+        # Clone related data
+        ReportSupplementaryVersionService.clone_report_version_operation(report_version, new_report_version)
+        ReportSupplementaryVersionService.clone_report_version_representatives(report_version, new_report_version)
+        ReportSupplementaryVersionService.clone_report_version_person_responsible(report_version, new_report_version)
+        ReportSupplementaryVersionService.clone_report_version_additional_data(report_version, new_report_version)
+        ReportSupplementaryVersionService.clone_report_version_new_entrant_data(report_version, new_report_version)
+        ReportSupplementaryVersionService.clone_report_version_verification(report_version, new_report_version)
+        ReportSupplementaryVersionService.clone_report_version_attachments(report_version, new_report_version)
+        ReportSupplementaryVersionService.clone_report_version_facilities(report_version, new_report_version)
+        # Return the newly created report version
+        return new_report_version
+
+    @staticmethod
+    def clone_report_version_operation(old_report_version: ReportVersion, new_report_version: ReportVersion) -> None:
+        # Get the old ReportOperation associated with the old report version
+        old_report_version_operation = ReportOperation.objects.get(report_version=old_report_version)
+        # Clone the ReportOperation for the new report version
+        new_report_version_operation = ReportOperation.objects.create(
+            operator_legal_name=old_report_version_operation.operator_legal_name,
+            operator_trade_name=old_report_version_operation.operator_trade_name,
+            operation_name=old_report_version_operation.operation_name,
+            operation_type=old_report_version_operation.operation_type,
+            registration_purpose=old_report_version_operation.registration_purpose,
+            operation_bcghgid=old_report_version_operation.operation_bcghgid,
+            bc_obps_regulated_operation_id=old_report_version_operation.bc_obps_regulated_operation_id,
+            report_version=new_report_version,
+        )
```

Comment:
Sorry I wasn't clear, I still think all these should be cloned with the official django method. This way we make this code resilient to us adding or removing fields from the different models

The part where we inspect the models dependencies of a report version is the one that can be done in tech debt 

Like:
```suggestion
        report_version_operation_to_clone = ReportOperation.objects.get(report_version=old_report_version)
        report_version_operation_to_clone.id = None
        report_version_operation_to_clone.report_version=new_report_version
        report_version_operation_to_clone.save()
```

Reply:
@pbastia thank-you, appreciate the learning- apologies for my misunderstanding, I will "make it right" now :)